src/freebsd.c

```diff
@@ -0,0 +1,71 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+#include <uv.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "../include/fs-ext.h"
+#include "platform.h"
+
+#ifndef RENAME_EXCHANGE
+#define RENAME_EXCHANGE 2
+#endif
+
+int
+fs_ext__try_lock (uv_os_fd_t fd, uint64_t offset, size_t length, fs_ext_lock_type_t type) {
+  if (offset != 0 || length != 0) return UV_EINVAL;
+
+  int res = flock(fd, (type == FS_EXT_WRLOCK ? LOCK_EX : LOCK_SH) | LOCK_NB);
+
+  return res == -1 ? uv_translate_sys_error(errno) : res;
+}
+
+int
+fs_ext__wait_for_lock (uv_os_fd_t fd, uint64_t offset, size_t length, fs_ext_lock_type_t type) {
+  if (offset != 0 || length != 0) return UV_EINVAL;
+
+  int res = flock(fd, type == FS_EXT_WRLOCK ? LOCK_EX : LOCK_SH);
+
+  return res == -1 ? uv_translate_sys_error(errno) : res;
+}
+
+int
+fs_ext__unlock (uv_os_fd_t fd, uint64_t offset, size_t length) {
+  if (offset != 0 || length != 0) return UV_EINVAL;
+
+  int res = flock(fd, LOCK_UN);
+
+  return res == -1 ? uv_translate_sys_error(errno) : res;
+}
+
+int
+fs_ext__trim (uv_os_fd_t fd, uint64_t offset, size_t length) {
+  int res = posix_fallocate(fd, offset, length);
+
+  return res == -1 ? uv_translate_sys_error(errno) : res;
+}
+
+int
+fs_ext__swap (const char *from, const char *to) {
+
+  char swap[1024]; // Max path length
+  char temp[8];
+
+  for(int i = 0; i < 8; i++) {
+    temp[i] = 'a' + (random() % 26);
```

Comment:
I guess this relies on someone already having provided a seed?

Reply:
Also, might it more appropriate to use `mktemp()`? https://www.freebsd.org/cgi/man.cgi?mktemp(3)